fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -367,6 +368,13 @@ private OptExpression logicalRuleRewrite(OptExpression tree, TaskContext rootTas
         return tree.getInputs().get(0);
     }
 
+    private OptExpression rewriteAndValidatePlan(OptExpression tree, TaskContext rootTaskContext) {
+        OptExpression result = logicalRuleRewrite(tree, rootTaskContext);
+        OptExpressionValidator validator = new OptExpressionValidator();
+        validator.visit(tree, null);
+        return result;
```

Comment:
better move it to last phase if you want to add a valid framework

Reply:
How about a two phase check? This check focuses on logical check. The last check focuses on physical check.

Reply:
It's unnecessary to check twice? and will easy to miss different check items in two checks?

Reply:
I think the reasons for the two-level validation are as follows. 1. The granularity of the validation is refined, which is helpful for troubleshooting. 2. The logic rewriting stage itself needs to ensure the accuracy of the logical information to avoid unexpected behavior caused by ref wrong logical proerpty in cbo optimize phase 3. The logic of the logical operator implementation is relatively simple, and the physical operator does not need to do the same validation logic. The validation of the physical stage should be combined with the build and distribute of the fragment to ensure the accuracy of the information delivered to be.